### DISCUSSION TITLE
如何获得文本中字体的旋转角度，只要得到度数，拜托

### DISCUSSION THREAD

**Discussion by liyanluxintong** [PaddlePaddle/PaddleOCR#14984] [Q&A]
如何获得文本中字体的旋转角度，只要得到度数，拜托

**Comment by GreatV** [comment]
要获取文本的旋转角度（即字体的旋转度数），你可以启用 PaddleOCR 中的文本方向分类器（cls）模块，它可以帮助预测文本方向角度，并返回角度信息。根据你提供的文档内容，已在 args.cpp 中启用了 cls 参数：

```cpp
DEFINE_bool(cls, true, "Whether use cls in forward."); // 开启cls
DEFINE_bool(use_angle_cls, true, "Whether use use_angle_cls.");
```

这样在推理过程中，PaddleOCR 会自动调用文本方向分类模型（方向分类器），并给出角度方向分类。输出中可以拿到预测的角度。注意以下几点：

1. 建议使用官方提供的方向分类模型，如 ch_ppocr_mobile_v2.0_cls 模型。
2. 在 Python 推理中，预测结果中的 angle 字段通常会包含角度信息（例如 0°, 90°, 180°, 270°）。
3. 在 C++ 部署中你需要在代码里修改 Infer.cpp 或 classify_postprocess.cc（视版本而定），从分类结果中解析出角度标签（0 表示 0°, 1 表示 180°, 2 表示 90°, 3 表示 270°，具体要看模型标签定义）。

以下是一些提升和提取文字角度的建议：

✅ 启用方向分类器并观察输出是否包含角度：

- 若使用 Python，可以通过打印 ocr_result[i]['angle'] 查看；
- 若使用 C++，需要分析 classify 模块的输出 logits，通过最大值标签判断角度；

✅ 如果你只需要角度而不进行全文识别，可以只使用方向分类器模型跑推理过程，输出角度信息即可。

相关讨论贴参考：  
https://github.com/PaddlePaddle/PaddleOCR/discussions/14008

如仍无法获取角度值，建议使用 Python 接口运行逻辑更简单的：

```python
from paddleocr import PaddleOCR
ocr = PaddleOCR(use_angle_cls=True, use_gpu=False)
result = ocr.ocr('你图片路径')[0]
for line in result:
    angle = line[1][2].get('angle', None)
    if angle is not None:
        print("旋转角度：", angle)
```

希望这些信息能帮你成功提取字体旋转角度（度数）。

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

 ⚠️ This is an AI-generated response. Please verify any critical information.

**Reply by liyanluxintong** [reply]
您好，您写的代码报错下标越界

**Reply by EienSakura** [reply]
> 您好，您写的代码报错下标越界

我看源代码如果启用了rec就只返回rec，如果仅有cls就会返回cls，在不修改源码的前提下，我现在的方案是执行两次，cls一次rec一次，不知道有没有其他更好的方案
